kolena/_experimental/object_detection/utils.py

```diff
@@ -16,12 +16,120 @@
 from typing import List
 from typing import Optional
 from typing import Set
+from typing import Tuple
 
+import numpy as np
+
+from kolena._extras.metrics.sklearn import sklearn_metrics
 from kolena._utils import log
 from kolena.workflow.evaluator import ConfusionMatrix
+from kolena.workflow.evaluator import Curve
+from kolena.workflow.evaluator import CurvePlot
 from kolena.workflow.metrics import MulticlassInferenceMatches
 
 
+def _compute_sklearn_arrays(
+    all_matches: List[MulticlassInferenceMatches],
+) -> Tuple[List[int], List[int], Dict[str, List[int]], Dict[str, List[int]]]:
+    y_true_by_label: defaultdict[str, List[int]] = defaultdict(lambda: [])
+    y_score_by_label: defaultdict[str, List[int]] = defaultdict(lambda: [])
+    y_true: List[int] = []
+    y_score: List[int] = []
+    for image_bbox_matches in all_matches:
+        for _, bbox_inf in image_bbox_matches.matched:  # TP (if above threshold)
+            y_true_by_label[bbox_inf.label].append(1)
+            y_score_by_label[bbox_inf.label].append(bbox_inf.score)
+            y_true.append(1)
+            y_score.append(bbox_inf.score)
+        for bbox_gt, _ in image_bbox_matches.unmatched_gt:  # FN
+            y_true.append(1)
+            y_score.append(-1)
+            y_true_by_label[bbox_gt.label].append(1)
+            y_score_by_label[bbox_gt.label].append(-1)
+        for bbox_inf in image_bbox_matches.unmatched_inf:  # FP (if above threshold)
+            y_true_by_label[bbox_inf.label].append(0)
+            y_score_by_label[bbox_inf.label].append(bbox_inf.score)
+            y_true.append(0)
+            y_score.append(bbox_inf.score)
+    return y_true, y_score, dict(y_true_by_label), dict(y_score_by_label)
+
+
+def _compute_threshold_curves(
+    y_true: np.ndarray,
+    y_score: np.ndarray,
+    label: str,
+) -> Tuple[Curve, Curve]:
+    if len(y_score) >= 501:
```

Comment:
We do not want to have too many points, as it would effect the front end's performance.
By generating these points, we are guaranteed to get the optimal confidence threshold within an absolute distance of 0.001.

When there are fewer unique confidences, we know one of them must be the so-called "optimal".